src/status_im/contexts/communities/overview/view.cljs

```diff
@@ -140,46 +121,39 @@
+        info-button-handler (fn []
+                              (rf/dispatch
+                               [:show-bottom-sheet
+                                {:content
+                                 (fn []
+                                   [quo/documentation-drawers
+                                    {:title        (i18n/label :t/token-gated-communities)
+                                     :show-button? true
+                                     :button-label (i18n/label :t/read-more)
+                                     :button-icon  :info}
+                                    [quo/text {:size :paragraph-2}
+                                     (i18n/label :t/token-gated-communities-info)]])}]))]
     (cond
-
       networks-not-supported?
       [network-not-supported]
 
       (or (not role-permissions?) no-member-permission?)
       [request-access-button id color]
 
       :else
-      [rn/view {:style (style/token-gated-container)}
-       [rn/view
-        {:style {:padding-horizontal 12
-                 :flex-direction     :row
-                 :align-items        :center
-                 :justify-content    :space-between
-                 :flex               1}}
-        [quo/text {:weight :medium}
-         (if (and can-request-access? highest-permission-role)
-           (i18n/label :t/you-eligible-to-join-as {:role highest-role-text})
-           (i18n/label :t/you-not-eligible-to-join))]
-        [info-button]]
-       [quo/text {:style {:padding-horizontal 12 :padding-bottom 6} :size :paragraph-2}
-        (if can-request-access?
-          (i18n/label :t/you-hodl)
-          (i18n/label :t/you-must-hold))]
-       [quo/token-requirement-list
-        {:tokens   tokens
-         :padding? true}]
-       [quo/button
-        {:on-press            (if config/community-accounts-selection-enabled?
-                                #(rf/dispatch [:open-modal :community-account-selection-sheet
-                                               {:community-id id}])
-                                #(rf/dispatch [:open-modal :community-requests-to-join {:id id}]))
-         :accessibility-label :join-community-button
-         :customization-color color
-         :container-style     {:margin-horizontal 12 :margin-top 8 :margin-bottom 12}
-         :disabled?           (not can-request-access?)
-         :icon-left           (if can-request-access? :i/unlocked :i/locked)}
-        (i18n/label :t/request-to-join)]])))
+      [quo/community-token-gating
+       {:role            highest-role-text
+        :tokens          tokens
+        :community-color color
+        :satisfied?      can-request-access?
+        :on-press        (if config/community-accounts-selection-enabled?
+                           #(rf/dispatch [:open-modal :community-account-selection-sheet
+                                          {:community-id id}])
+                           #(rf/dispatch [:open-modal :community-requests-to-join
```

Comment:
We can cache these functions with `use-callback`, especially because they only depend on the `id`, which basically won't ever change. This should help Reagent not needlessly process `community-token-gating`.